src/ln/functional_tests.rs

```diff
@@ -5352,3 +5352,261 @@ fn test_update_fulfill_htlc_bolt2_after_malformed_htlc_message_must_forward_upda
 
 	check_added_monitors!(nodes[1], 1);
 }
+
+#[test]
+fn test_commitment_signed_bolt2_sending_node_must_not_send_commitment_signed_no_updates() {
+	//BOLT 2 Requirement: A sending node:
+	//    * MUST NOT send a commitment_signed message that does not include any updates
+
+	let nodes = create_network(2);
+	let (_, _, channel_id, _) = create_announced_chan_between_nodes(&nodes, 0, 1);
+
+	let mut chan_state_lock = nodes[0].node.channel_state.lock().unwrap();
+	let chan_state = chan_state_lock.borrow_parts();
+
+	let chan = chan_state.by_id.get_mut(&channel_id).unwrap();
+	if let Err(super::channel::ChannelError::Ignore(msg)) = chan.send_commitment() {
+		assert_eq!(msg, "Cannot create commitment tx until we have some updates to send");
+	} else {
+		panic!("send_commitment did not return expected error with no updates")
+	}
+}
+
+#[test]
+fn test_commitment_signed_bolt2_sending_node_may_send_commitment_signed_new_revocation_number() {
+	//BOLT 2 Requirement: A sending node:
+	//    * MAY send a commitment_signed message that doesn't change the commitment transaction aside from the new revocation number (due to dust, identical HTLC replacement, or insignificant or multiple fee changes).
+
+	const VALUE_DUST: u64 = 10000;
+
+	let mut nodes = create_network(2);
+	let (_, _, channel_id, _) = create_announced_chan_between_nodes_with_value(&nodes, 0, 1, VALUE_DUST, 10001);
+	route_payment(&nodes[0], &[&nodes[1]], VALUE_DUST);
+
+	let node_a_output_value = {
+		let chan_lock = nodes[0].node.channel_state.lock().unwrap();
+		let last_local_commitment_txn = &chan_lock.by_id.get(&channel_id).unwrap().last_local_commitment_txn;
+		last_local_commitment_txn[0].output[0].value
+	};
+
+	let node_b_output_value = {
+		let chan_lock = nodes[1].node.channel_state.lock().unwrap();
+		let last_local_commitment_txn = &chan_lock.by_id.get(&channel_id).unwrap().last_local_commitment_txn;
+		last_local_commitment_txn[0].output[0].value
+	};
+
+	assert!(node_a_output_value < VALUE_DUST);
+	assert_eq!(node_a_output_value, node_b_output_value);
```

Comment:
Alice and Bob's commitment transactions are never _structurally_ no more_ physically_ equal. Following bolt5 - Commitment Transaction,  they are symmetrical, i.e script encumbering outputs should produce same incentives (if I get it right). So what you want there is testing Alice previous commitment tx against Alice last commitment tx, just mask locktime (in which revocation number is encoded)